File: proselint/checks/hedging.py
"""
Hedging.

---
layout:     post
source:     Pinker's book on writing
source_url: ???
title:      hedging
date:       2014-06-10 12:31:19
categories: writing
---

Points out hedging.

"""

from proselint.registry.checks import Check, types

check = Check(
    check_type=types.Existence(
        items=(
            "I would argue that",
            ", so to speak",
            "to a certain degree",
        )
    ),
    path="hedging",
    message="Hedging. Just say it.",
)

__register__ = (check,)

File: proselint-rs/crates/proselint-checks/src/hedging.rs
use proselint_registry::checks::{Check, CheckType, Padding};

pub const EXAMPLES_PASS: &[&str] = &["Smoke phrase with nothing flagged."];
pub const EXAMPLES_FAIL: &[&str] = &[
    "I would argue that this is a good test.",
    "You could say that, so to speak.",
];

const CHECK: Check = Check {
	check_type: CheckType::Existence {
		items: &[
			"I would argue that",
			"so to speak",
			"to a certain degree",
		],
		padding: Padding::WordsInText,
		exceptions: &[],
	},
	path: "hedging.misc",
	msg: "Hedging. Just say it.",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];
